frappe/contacts/doctype/address/address.py

```diff
@@ -145,30 +145,10 @@ def get_list_context(context=None):
 def get_address_list(doctype, txt, filters, limit_start, limit_page_length = 20, order_by = None):
 	from frappe.www.list import get_list
 	user = frappe.session.user
-	ignore_permissions = False
-	if is_website_user():
-		if not filters: filters = []
-		add_name = []
-		contact = frappe.db.sql("""
-			select
-				address.name
-			from
-				`tabDynamic Link` as link
-			join
-				`tabAddress` as address on link.parent = address.name
-			where
-				link.parenttype = 'Address' and
-				link_name in(
-				   select
-					   link.link_name from `tabContact` as contact
-				   join
-					   `tabDynamic Link` as link on contact.name = link.parent
-				   where
-					   contact.user = %s)""",(user))
-		for c in contact:
-			add_name.append(c[0])
-		filters.append(("Address", "name", "in", add_name))
-		ignore_permissions = True
+	ignore_permissions = True
+
+	if not filters: filters = []
+	filters.append(("Address", "owner", "=", user))
```

Comment:
You can add a Role Permission for "All" role with if_owner check. It will the same effect as you are trying to do here.

<img width="1127" alt="Screenshot 2019-11-21 at 9 05 23 AM" src="https://user-images.githubusercontent.com/13928957/69302043-4bf3dc80-0c3e-11ea-9188-f50216e520d7.png">

Reply:
wouldn't that also affect desk users then? this change is only specific to the standard address web form

Reply:
Yes, all desk users will be able to see addresses that they have created. I don't see any problem with this.